grib/src/main/java/ucar/nc2/grib/collection/GribIosp.java

```diff
@@ -786,33 +786,39 @@ private Array makeLazyTime2Darray(Variable coord, Time2Dinfo info) {
 
       case intvU:
         // data[nruns*ntimes]
+        double[] currentBounds = new double[2];
+        double[] prevBounds = new double[2];
         for (int runIdx = 0; runIdx < nruns; runIdx++) {
           CoordinateTimeIntv timeIntv = (CoordinateTimeIntv) time2D.getTimeCoordinate(runIdx);
-          double valueBound1 = Double.NaN;
-          double valueBound2 = Double.NaN;
-          double twicePrevValueBound2 = Double.NaN;
+          int timeUnitValue = timeUnit.getValue();
+          int time2D_Offset = time2D.getOffset(runIdx);
           int timeIndex = 0;
           for (TimeCoordIntvValue tinv : timeIntv.getTimeIntervals()) {
-            valueBound1 = timeUnit.getValue() * tinv.getBounds1() + time2D.getOffset(runIdx);
-            valueBound2 = timeUnit.getValue() * tinv.getBounds2() + time2D.getOffset(runIdx);
+            currentBounds[0] = timeUnitValue * tinv.getBounds1() + time2D_Offset;
+            currentBounds[1] = timeUnitValue * tinv.getBounds2() + time2D_Offset;
             // Use end-point of current interval as initial guess for current time coordinate value
-            data[timeIndex] = valueBound2;
+            data[timeIndex] = currentBounds[1];
```

Comment:
I think that `data` is sort of a 2D array flattened into a 1D, with size `nruns*ntimes` (I see mentioned in the comment above). So maybe its index should be `runIdx * ntimes + timeIndex` instead of `timeIndex`

Reply:
Yes, definitely. Thanks! I knew I had to revisit the 2D time case but forgot to even allow for it. I'll get on it, tomorrow.

Reply:
Fixed by switching to an index from zero to `nruns*ntimes` and a run offset to help with checking the pseudo array bounds. Tests are good locally. Added a few that depend on `cdmUnitTest` data.